scipy/interpolate/fitpack/splint.f

```diff
@@ -41,9 +43,9 @@ recursive function splint(t,n,c,k,a,b,wrk) result(splint_res)
 c
 c  ..scalar arguments..
       real*8 a,b
-      integer n,k
+      integer n,k, nc
 c  ..array arguments..
-      real*8 t(n),c(n),wrk(n)
+      real*8 t(n),c(nc),wrk(n)
```

Comment:
To verify that this, compare to the  `do i=1, nk1` loop below, where `nk1=n-k-1`. This way, the elemnts of the `c` array with indices past `n-k-1` are not referenced even though the array is declated to be of size `n`.